Source/cryptography/ICryptography.h

```diff
@@ -103,6 +106,35 @@ namespace Cryptography {
         virtual uint32_t Derive(const uint32_t privateKey, const uint32_t peerPublicKeyId, uint32_t& secretId /* @out */) = 0;
     };
 
+    struct IPersistent : virtual public Core::IUnknown {
+
+    enum { ID = ID_PERSISTENT };
+
+    enum keytype {
+        AES128,
+        AES256,
+        HMAC128,
+        HMAC160,
+        HMAC256
+    };
+
+    virtual ~IPersistent() { }
+
+    //Check if a named key exists in peristent storage
+    virtual uint32_t Exists(const string& locator, bool& result) const =0;
+
+    //Load persistent key details to vault
+    virtual uint32_t Load(const string& locator, uint32_t&  id) = 0;
+
+    //Create a new key on persistent storage
+    virtual uint32_t Create(const string& locator, const keytype keyType, uint32_t& id) = 0 ;
+
+    //To explicitly flush resources at the backend
+    virtual uint32_t Persistent_Flush() = 0;
```

Comment:
Why do you re-iterate on the Persistent ? The Flush is already part of the Scope IPersistent so Indeed it will be a Persistent_Flush :-) Wouldn't it be more convenient not to re-iterate on the parent (interface name) and just call this:
virtual uint32_t Flush() = 0;